sdk/core/core/test/cmocka/test_span.c

```diff
@@ -288,6 +296,56 @@ static void az_span_find_error_cases_fail()
   assert_int_equal(az_span_find(AZ_SPAN_NULL, target), -1);
 }
 
+static void az_span_find_target_longer_than_source_fails()
+{
+  az_span span = AZ_SPAN_FROM_STR("aa");
+  az_span target = AZ_SPAN_FROM_STR("aaa");
+
+  assert_int_equal(az_span_find(span, target), -1);
+}
+
+static void az_span_find_target_overlap_continuation_of_source_fails()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcd");
+  az_span target = AZ_SPAN_FROM_STR("cde");
+
+  assert_int_equal(az_span_find(span, target), -1);
+}
+
+static void az_span_find_target_more_chars_than_preffix_of_source_fails()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcd");
+  az_span target = AZ_SPAN_FROM_STR("zab");
+
+  assert_int_equal(az_span_find(span, target), -1);
+}
+
+static void az_span_find_overlaping_target_success()
```

Comment:
nit: Since we are doing overlapping tests, can we add one where the source and target both come as a slice of a larger buffer and that the find fails (returns -1)?

```c
  az_span span = AZ_SPAN_FROM_STR("abcdefghij");
  az_span source = az_span_slice(span, 1, 4);
  az_span target = az_span_slice(span, 6, 9);
  assert_int_equal(az_span_find(source, target), -1);
  assert_int_equal(az_span_find(source, az_span_slice(span, 1, 5)), -1);
  assert_int_equal(az_span_find(source, az_span_slice(span, 2, 4)), -1);
```

Also typo:
```suggestion
static void az_span_find_overlapping_target_success()
```

Reply:
Yes, added. 
Note that "assert_int_equal(az_span_find(source, az_span_slice(span, 2, 4)), -1);" is not true, the `target` would be contained in `source`. The correct would be to expect return equal 1.

And typo fixed.

Reply:
> Note that "assert_int_equal(az_span_find(source, az_span_slice(span, 2, 4)), -1);" is not true, the `target` would be contained in `source`.

You are right, that's what I meant.